lib/bosh/BoshDirectorClient.js

```diff
@@ -374,7 +373,9 @@ class BoshDirectorClient extends HttpClient {
 
   }
 
-  pollTaskStatusTillComplete(taskId, pollInterval, timeout) {
+  pollTaskStatusTillComplete(taskId, pollInterval, timeout, maxErrorRetry) {
+    let errorRetries = 0;
+    maxErrorRetry = maxErrorRetry || 3;
```

Comment:
The default value 3 is hard coded, is it better to to put in constant so that somewhere can be used also .. or configurable.
so that in similar cases can be used!